### PR TITLE
Qol 8542 improve loading on the search results page global search

### PR DESCRIPTION
**Pull request by asifamingov**

### CODE REVIEW

**Review by ThrawnCA — APPROVED** [review]

File: global-search/src/modules/SearchModule.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private readonly urlParameter: any;
+
+    constructor() {
+        this.urlParameter = urlParameterMap();
+        this.funnelbackApiUrl = 'https://find.search.qld.gov.au/s/search.json';
+        this.spinnerEl = document.querySelector(".qg-search-results__spinner");
+        this.siteInput = document.querySelector('.qg-site-search__component .qg-search-site__input');
+
+        // check if query porameter is set to start fetch process
+        let queryParam = this.urlParameter.query;
```
Comment (devonpis) [inline global-search/src/modules/SearchModule.ts:19]: better use `const` if variable will not change again

File: global-search/src/modules/FilterModule.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * filterResultsTemplate display a template to filter search results
+     * @return {undefined}
+     * */
+    filterResultsTemplate(){
+        let profile = this.urlParameter.profile;
+
+        if(this.urlParameter.filter || profile && profile !== 'qld' || this.urlParameter.scope){
+            render(filterResultsTemplate(), document.getElementById('qg-filter-by-results') as HTMLBodyElement);
```
Comment (devonpis) [inline global-search/src/modules/FilterModule.ts:39]: not sure if type assertion is needed here, maybe check `document.getElementById('qg-filter-by-results')` is not undefined first?

File: global-search/src/template/filter-results.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import {html, render} from 'lit-html';
+import {mainTemplate} from './main';
+import {urlParameterMap} from '../utils/urlParameter';
+import {fetchData} from '../utils/fetchData';
+
+export function filterResultsTemplate() {
+    let label: string = '';
```
Comment (devonpis) [inline global-search/src/template/filter-results.ts:7]: seems most of the variables here could be `const`

**Review by devonpis — APPROVED** [review]
just a note for the future, would be nice if eslint/tslint is integrated in this project so most of the linting issues could be found automatically.

**Comment by asifamingov** [comment]
> just a note for the future, would be nice if eslint/tslint is integrated in this project so most of the linting issues could be found automatically.

Added ESlint

File: dist/index.c2b3c8ae.css [hunk 4]
```diff
@@ -0,0 +1,71 @@
+#qg-two-col-aside.wide #qg-primary-content {
```
Comment (duttonw) [inline dist/index.c2b3c8ae.css:1]: I'm really not a fan of having 'dist' binaries commited into a source repo.

@asifamingov , please raise a ticket for making a matrix-projects-bin repo and cloning the formio repo pipeline for 'fleshing' it out via github action calls as well as make progression to reference that repo in squiz matrix etc.

**Review by duttonw — APPROVED** [review]
I'm really digging the usage of typescript for source programming.  

Do look at raising that improvement i've commented about 'dist' folder/s.

File: dist/index.c2b3c8ae.css (hunk 4 above)
Comment (asifamingov) [reply dist/index.c2b3c8ae.css:1]: 'Dist' folder is temporary I will create a Gitbridge.

**Pull request merged**